### PR TITLE
Add conduit-pipeline-check utility

### LINKED ISSUE #773: Feature: command line tool for health checking pipelines
### Feature description

Command line tool which checks the health of an existing pipeline. 
The tool will return non-zero status when the pipeline is not running or does not exist.

It can be used in automation systems, such as kubernetes to health check conduit deployments.

### PR DESCRIPTION
**Pull request by lyuboxa**
### Description

The command line utility allows users or automation to request status of a pipeline and return a non-zero code when the pipeline is not running. The check happens over gRPC.

Additionally include a .dockerignore file to speed up docker builds.

Fixes #773 

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/ConduitIO/conduit/pulls) for the same update/change.
- [ ] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: cmd/conduit-pipeline-check/main.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+	opts := []grpc.DialOption{
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+	}
+
+	dialCtx, dialCancel := context.WithTimeout(ctx, *timeout)
+	defer dialCancel()
+
+	c, err := grpc.DialContext(dialCtx, *grpcAddress, opts...)
+	if err != nil {
+		return exitCodeErr, fmt.Errorf("failed to connect to conduit grpc server: %v", err)
```
Comment (hariso) [inline cmd/conduit-pipeline-check/main.go:70]: Nitpick: %v should be %w

File: cmd/conduit-pipeline-check/main.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	defer c.Close()
+
+	pipeline := apiv1.NewPipelineServiceClient(c)
+	p, err := pipeline.GetPipeline(
+		ctx,
+		&apiv1.GetPipelineRequest{
+			Id: *pipelineName,
+		})
+	if err != nil {
+		return exitCodeErr, fmt.Errorf("failed to find pipeline %q: %v", *pipelineName, err)
```
Comment (hariso) [inline cmd/conduit-pipeline-check/main.go:81]: %v -> %w

Comment (lyuboxa) [reply cmd/conduit-pipeline-check/main.go]: fixed in [5cb1a8](https://github.com/ConduitIO/conduit/pull/774/commits/75cb1a895fb83bebefc230d4ef24f7d73e4ab118)

File: cmd/conduit-pipeline-check/main.go (hunk 1 above)
Comment (lyuboxa) [reply cmd/conduit-pipeline-check/main.go]: fixed in [5cb1a8](https://github.com/ConduitIO/conduit/pull/774/commits/75cb1a895fb83bebefc230d4ef24f7d73e4ab118)

File: cmd/conduit-pipeline-check/main.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+// * missing or invalid flags
+// * gRPC connection failures
+// * operation takes too long to execute
+func checkPipeline(ctx context.Context) (int, error) {
+	flags := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	var (
+		grpcAddress  = flags.String("grpc.address", ":8084", "address of the Conduit server gRPC API")
+		pipelineName = flags.String("pipeline.name", "", "name of the target pipeline")
+		timeout      = flags.Duration("timeout", 10*time.Second, "timeout duration")
+		verbose      = flags.Bool("verbose", false, "return non-zero code on failure instead of logging")
```
Comment (lovromazgon) [inline cmd/conduit-pipeline-check/main.go:57]: IIUC `verbose` will make sure a log is printed _in addition_ to the exit code, right? The message seems a bit misleading.

File: cmd/conduit-pipeline-check/main.go [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+)
+
+func main() {
+	ctx := cancelOnInterrupt(context.Background())
+
+	exitcode, err := checkPipeline(ctx)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+	}
+	os.Exit(exitcode)
```
Comment (lovromazgon) [inline cmd/conduit-pipeline-check/main.go:43]: Here `exitcode` will always be `exitCodeErr` if an error is returned or `exitCodeOk` if it's not, right? We could simplify this:
```suggestion
	err := checkPipeline(ctx)
	if err != nil {
		fmt.Fprintln(os.Stderr, err)
		os.Exit(exitCodeErr)
	}
```
The nice thing about this is we don't call `os.Exit(0)` which just terminates the program right away without calling any deferred functions. Not that it matters here, but you know, good practice 🙃

**Review by lovromazgon — APPROVED** [review]
LGTM, just a few nitpicks 😉

File: cmd/conduit-pipeline-check/main.go (hunk 4 above)
Comment (lyuboxa) [reply cmd/conduit-pipeline-check/main.go:43]: yup this makes sense. will make the change, originally i started with just an error as well.

File: cmd/conduit-pipeline-check/main.go (hunk 3 above)
Comment (lyuboxa) [reply cmd/conduit-pipeline-check/main.go:57]: ugh yea, will update

File: cmd/conduit-pipeline-check/main.go (hunk 4 above)
Comment (lyuboxa) [reply cmd/conduit-pipeline-check/main.go]: fixed in bc01d7c425731ccb18893dd3447adcec34012f64

**Pull request merged**